core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -66,6 +70,28 @@ private void preparePartitionedTable() {
         .commit();
   }
 
+  @Test
+  public void testManifestsTableWithDroppedPartition() throws IOException {
+    table.newFastAppend()
+        .appendFile(FILE_A)
+        .appendFile(FILE_B)
+        .commit();
+
+    table.updateSpec().removeField(Expressions.bucket("data", 16)).commit();
```

Comment:
Could we add a test which drops and adds partitioning more than once? I just want to have confidence for something like

Int -> Void<String> ->  Int -> Void<String> -> Truncate<String>